sklearn/ensemble/losses.py

```diff
@@ -0,0 +1,890 @@
+"""Losses and corresponding default initial estimators for gradient boosting
```

Comment:
Is this move mostly cosmetic? It seems a bit strange to have it sit under `ensemble` without qualifying it as related to GBDT

Reply:
Please confirm that the only changes here are `get_init_raw_predictions` and the init_estimator (I know, I can diff)

Reply:
These are the most important changes but also this in losses.py the losses are private, so we can change them how we please.

gradient_boosting.py was is 2500 lines long and I think it really benefits from having the losses in a separate file. Makes reading / editing the code easier.

Reply:
(Also I fixed a few docstrings)